### ISSUE TITLE
[REVIEW]: DeGAUSS: Decentralized Geomarker Assessment for Multi-Site Studies

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#812]
**Submitting author:** @cole-brokamp (<a href="http://orcid.org/0000-0002-0289-3151">Cole Brokamp</a>)
**Repository:** <a href="https://github.com/cole-brokamp/DeGAUSS" target ="_blank">https://github.com/cole-brokamp/DeGAUSS</a>
**Version:** v0.2
**Editor:** @pjotrp
**Reviewer:** @vsoch, @george-githinji
**Archive:** <a href="https://doi.org/10.5281/zenodo.1443521" target="_blank">10.5281/zenodo.1443521</a>

## Status

[![status](http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81/status.svg)](http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81"><img src="http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81/status.svg)](http://joss.theoj.org/papers/51cadcd9bce9b42b6164e71cf708eb81)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/cole-brokamp/DeGAUSS" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@vsoch & @george-githinji, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @pjotrp know.

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/cole-brokamp/DeGAUSS">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v0.2)?
- [x] **Authorship:** Has the submitting author (@cole-brokamp) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/cole-brokamp/DeGAUSS">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (v0.2)?
- [x] **Authorship:** Has the submitting author (@cole-brokamp) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks. @vsoch, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[--> Check article proof :page_facing_up: <--](https://github.com/openjournals/joss-papers/blob/joss.00812/joss.00812/10.21105.joss.00812.pdf)

**Comment by pjotrp** [comment]
@cole-brokamp we are starting review! To expedite matters do you mind going through above check list and make sure the reviewers can tick the boxes? You can't check them yourself, but you can make sure they are covered.

**Comment by vsoch** [comment]
here are updated review guidelines : https://joss.readthedocs.io/en/latest/review_criteria.html

**Comment by cole-brokamp** [comment]
Thanks, @pjotrp. I believe that the manuscript and repository checks all the boxes.  Looking forward to hearing from the reviewers.

**Comment by vsoch** [comment]
@cole-brokamp @pjotrp I am starting my review, will post shortly!

**Comment by vsoch** [comment]
I have finished my review, reported in issue here: https://github.com/cole-brokamp/DeGAUSS/issues/10 I will go through the points in this repository and confirm when the issues have been addressed!

**Comment by pjotrp** [comment]
@cole-brokamp What is the latest news here?

**Comment by cole-brokamp** [comment]
I’m still working on my revisions.  Is there a deadline I should be aware of?  Thanks for your patience.

**Comment by pjotrp** [comment]
No deadlines - just interest :)

**Comment by cole-brokamp** [comment]
OK, great.  Thanks for your patience — grant season is almost finished and then I’ll be there!

**Comment by vsoch** [comment]
I'll be here too!!

**Comment by cole-brokamp** [comment]
I've revised the manuscript, code, and README based on @vsoch's review.  You can see a summary of my changes here: https://github.com/cole-brokamp/DeGAUSS/issues/10#issuecomment-419918139

**Comment by cole-brokamp** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Compile the paper
@whedon generate pdf

🚧 🚧 🚧 Experimental Whedon features 🚧 🚧 🚧

# Compile the paper from a custom git branch
@whedon generate pdf from branch custom-branch-name

```

**Comment by cole-brokamp** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.00812/joss.00812/10.21105.joss.00812.pdf)

**Comment by vsoch** [comment]
Bravo @cole-brokamp! You have done a beautiful job fixing up the paper and repository. I have two quick checks to do before signing off, both rather trivial.

- tests: do you describe anywhere how to test? Are there automated tests of any kind? The general goal would be that if a developer stumbles across any of the code and wants to quickly check "is this functioning as it should on my system" you can point them to the things to run.
 - paper rending: I noticed there seems to be a missing / not rendered double slash before "a process"

![image](https://user-images.githubusercontent.com/814322/45324890-77007d80-b51d-11e8-8339-821931eecb66.png)

You might not have any control over this, but it might be worth glancing over. Those are the only two remaining (small) things I would want to ask about before signing off.

**Comment by cole-brokamp** [comment]
I’m working on adding some automated build testing to address the last comment. I’ll let you know when it’s ready for a final review.

**Comment by vsoch** [comment]
+1! We had good discussion here https://github.com/cole-brokamp/DeGAUSS/issues/10#issuecomment-421098449 about tests and @cole-brokamp is tackling this probably as we speak! And I’m ready at bay for when Cole needs me!

**Comment by cole-brokamp** [comment]
some delays on getting time to work on this (as usual ...), but its coming soon!

**Comment by pjotrp** [comment]
No worries, as long as we get there :)

**Comment by cole-brokamp** [comment]
I've addressed the last comment by implementing tests (see https://github.com/cole-brokamp/DeGAUSS/issues/10#issuecomment-426444872 for more details).  What do you think, @vsoch?

**Comment by cole-brokamp** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.00812/joss.00812/10.21105.joss.00812.pdf)

**Comment by vsoch** [comment]
@cole-brokamp has done a fantastic job adding tests (and we are green and good to go!) and so I've updated my review and am very happy to say that @cole-brokamp has addressed all of my concerns, and I my recommendation is accept for publication in Joss. Great job @cole-brokamp! Passing forward to you again @pjotrp. 

https://travis-ci.org/cole-brokamp/DeGAUSS

**Comment by pjotrp** [comment]
The review process is now complete. To finalize your submission and accept your paper in JOSS, we need two things. First, can you confirm that all references in your bibliography have a DOI (if one exists). Mind, it looks OK to me.

Second, we need you to deposit a copy of your software repository (including any revisions made during the JOSS review process) with a data-archiving service. To do so:

 1. Create [a GitHub release](https://help.github.com/articles/creating-releases/) of the current version of your software repository
 2. Deposit that release with [Zenodo](https://zenodo.org/), [figshare](https://figshare.com/), or a similar DOI issuer.
 3. Post a comment here with the DOI for the release.

**Comment by cole-brokamp** [comment]
Most of my references are either software vignettes or United States Public Law, both of which do not have DOIs available.  I've included all other DOIs that are available for my references.

I have tagged a release as v0.3 that includes all changes made during the JOSS review.  I have an automated build setup on Zenodo for all releases, so the current DOI will point to the latest version: 10.5281/zenodo.570873

I'm waiting on the Zenodo automated archiving for v0.3, which will be the release linked to the JOSS publication -- I'll update here when the DOI pointing specifically to v0.3 is available.

**Comment by cole-brokamp** [comment]
Here is the DOI for the v0.3 release:

10.5281/zenodo.1443521

Thanks again

**Comment by vsoch** [comment]
Don't forget the cool zenodo badge! 😎

**Comment by cole-brokamp** [comment]
no way I could forget that! 😉

**Comment by pjotrp** [comment]
@whedon set 10.5281/zenodo.1443521 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.1443521" target="_blank">10.5281/zenodo.1443521</a> is the archive.

**Comment by pjotrp** [comment]
@arfon we are ready to publish!

**Comment by vsoch** [comment]
Woohoo! Go @whedon bot go! :champagne:

**Comment by arfon** [comment]
@vsoch, @george-githinji - many thanks for your reviews here and to @pjotrp for editing this submission ✨ 

@cole-brokamp - your paper is now accepted into JOSS and your DOI is https://doi.org/10.21105/joss.00812 :zap: :rocket: :boom:

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](http://joss.theoj.org/papers/10.21105/joss.00812/status.svg)](https://doi.org/10.21105/joss.00812)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.00812">
  <img src="http://joss.theoj.org/papers/10.21105/joss.00812/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: http://joss.theoj.org/papers/10.21105/joss.00812/status.svg
   :target: https://doi.org/10.21105/joss.00812
```
This is how it will look in your documentation:

[![DOI](http://joss.theoj.org/papers/10.21105/joss.00812/status.svg)](https://doi.org/10.21105/joss.00812)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: http://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://www.flipcause.com/secure/cause_pdetails/Mjk3Nzk=

**Comment by cole-brokamp** [comment]
Thanks everyone for their help in getting this manuscript published.  It has been a great experience and I look forward to reviewing for JOSS soon.
